sns/client.bal

```diff
@@ -0,0 +1,501 @@
+                                     @display {label: "Target ARN"}  string? targetArn = (), 
+                                     @display {label: "Message Subject"} string? subject = (), 
+                                     @display {label: "Phone Number"} string? phoneNumber = (), 
+                                     @display {label: "Message Structure"} string? messageStructure = (), 
+                                     @display {label: "Message Duplication Id"} string? messageDeduplicationId = (), 
+                                     @display {label: "Message Group Id"} string? messageGroupId = (), 
+                                     @display {label: "Message Attributes"} MessageAttribute? messageAttributes = ()) 
+                                     returns @tainted @display {label: "Published result"} PublishResponse|error {
+        map<string> parameters = {};
+        parameters = createQueryString("Publish", parameters);
+        parameters["Message"] = message;
+        parameters = check addPublishOptionalParameters(parameters, topicArn, targetArn, subject, phoneNumber, messageStructure, messageGroupId, messageDeduplicationId, messageAttributes);
+        http:Request|error request = self.generateRequest(self.createPayload(parameters));
+        xml response = check sendRequest(self.amazonSNSClient, request);
+        PublishResponse|error publishResponse = xmlToPublishResponse(response);
+        if (publishResponse is PublishResponse) {
+            return publishResponse;
+        } else {
+            return error(publishResponse.message());
+        }
+    }
+
+    # Unsubscribe to a topic.
+    #
+    # + subscriptionArn - The ARN of the subscription
+    # + return - Result of unsubscription on success else an `error`
+    @display {label: "Unsubscribe Topic"}
+    isolated remote function unsubscribe(@display {label: "Subscription ARN"} string subscriptionArn) 
+                                         returns @tainted @display {label: "Unsubscription Status"} UnsubscribeResponse|error {
+        map<string> parameters = {};
+        parameters = createQueryString("Unsubscribe", parameters);
+        parameters["SubscriptionArn"] = subscriptionArn;
+        http:Request|error request = self.generateRequest(self.createPayload(parameters));
+        xml response = check sendRequest(self.amazonSNSClient, request);
+        UnsubscribeResponse|error unsubscribeResponse = xmlToUnsubscribeResponse(response);
+        if (unsubscribeResponse is UnsubscribeResponse) {
+            return unsubscribeResponse;
+        } else {
+            return error(unsubscribeResponse.message());
+        }
+    }
+
+    # Delete a topic.
+    #
+    # + topicArn - The ARN of the topic
+    # + return - Result of deleted topic on success else an `error`
+    @display {label: "Delete Topic"}
+    isolated remote function deleteTopic(@display {label: "Topic ARN"} string topicArn) 
```

Comment:
Say I want to delete an existing topic. How can I get the `topicArn` of the topic I want to delete? I might know only the topic name.